internal/fileutils/fileutils_test.go

```diff
@@ -0,0 +1,144 @@
+package fileutils_test
+
+import (
+	"os"
+	"path/filepath"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+	"github.com/ubuntu/authd-oidc-brokers/internal/fileutils"
+)
+
+func TestFileExists(t *testing.T) {
+	tests := map[string]struct {
+		name            string
+		fileExists      bool
+		parentDirIsFile bool
+
+		expectExists bool
+		expectError  bool
+	}{
+		"Returns true when file exists":                      {fileExists: true, expectExists: true},
+		"Returns false when file does not exist":             {fileExists: false, expectExists: false},
+		"Returns false when parent directory does not exist": {fileExists: false, expectExists: false},
+
+		"Error when parent directory is a file": {parentDirIsFile: true, expectError: true},
+	}
+
+	for _, tc := range tests {
+		t.Run(tc.name, func(t *testing.T) {
+			tempDir := t.TempDir()
+			path := filepath.Join(tempDir, "file")
+			if tc.fileExists {
+				err := fileutils.Touch(path)
+				require.NoError(t, err, "Touch should not return an error")
+			}
+			if tc.parentDirIsFile {
+				path = filepath.Join(tempDir, "file", "file")
+				err := fileutils.Touch(filepath.Join(tempDir, "file"))
+				require.NoError(t, err, "Touch should not return an error")
+			}
+
+			exists, err := fileutils.FileExists(path)
+			if tc.expectError {
+				require.Error(t, err, "FileExists should return an error")
+			} else {
+				require.NoError(t, err, "FileExists should not return an error")
+			}
```

Comment:
As we talked about in the HO: return early if expecting the error instead of the else.

```suggestion
			if tc.expectError {
				require.Error(t, err, "FileExists should return an error")
				return
			} 
			require.NoError(t, err, "FileExists should not return an error")
```

Reply:
done